Documentation/intro-to-clr.md

```diff
@@ -35,55 +35,55 @@ Now that we have basic idea what the CLR is, it is useful to back up just a bit
+Still, it is not the usefulness of garbage collection that makes it worthy of special note here.  More important is the simple requirement it places on the runtime itself:
 
 > Garbage collection requires ALL references to the GC heap to be tracked.
 
-While this is a very simple requirement, it in fact has profound ramifications for the runtime. As you can imagine, knowing where every pointer to an object is at every moment of program execution can be quite difficult. We have one mitigating factor, though. Technically, this requirement only applies to when a GC actually needs to happen (thus, in theory we don't need to know where all GC references are all the time, but only at the time of a GC). In practice, however, this mitigation doesn't completely apply because of another feature of the CLR:
+While this is a very simple requirement, it in fact has profound ramifications for the runtime.  As you can imagine, knowing where every pointer to an object is at every moment of program execution can be quite difficult.  We have one mitigating factor, though.  Technically, this requirement only applies to when a GC actually needs to happen (thus, in theory we don't need to know where all GC references are all the time, but only at the time of a GC).  In practice, however, this mitigation doesn't completely apply because of another feature of the CLR:
 
 > The CLR supports multiple concurrent threads of execution with a single process.
 
-At any time some other thread of execution might do an allocation that needs to do a garbage collection. The exact sequence of operations across concurrently executing threads is non-deterministic. We can't tell exactly what one thread will be doing when another thread requests an allocation that will trigger a GC. Thus, GCs can really happen any time.  Now the CLR does NOT need to respond _immediately_ to another thread's desire to do a GC, so the CLR has a little "wiggle room" and doesn't need to track GC references at _all_ points of execution, but it _does_ need to do so at enough places that it can guarantee "timely" response to the need to do a GC caused by an allocation on another thread.
+At any time some other thread of execution might perform an allocation requires a garbage collection.  The exact sequence of operations across concurrently executing threads is non-deterministic.  We can't tell exactly what one thread will be doing when another thread requests an allocation that will trigger a GC.  Thus, GCs can really happen any time.  Now the CLR does NOT need to respond _immediately_ to another thread's desire to do a GC, so the CLR has a little "wiggle room" and doesn't need to track GC references at _all_ points of execution, but it _does_ need to do so at enough places that it can guarantee "timely" response to the need to do a GC caused by an allocation on another thread.
```

Comment:
`might perform an allocation requires a garbage collection` <-- missing a "that" in there. LGTM otherwise.